### ISSUE TITLE
Accessibility issue: Cannot use aria-label or aria-labelledby on elements and roles that prohibit naming

### ISSUE THREAD

**Issue by shaunnajk** [fullcalendar/fullcalendar#7566]
### Reduced Test Case

https://stackblitz.com/edit/github-cxlana-ekghpf?file=src%2FDemoApp.jsx

### Do you understand that if a reduced test case is not provided, we will intentionally delay triaging of your ticket?

- [X] I understand

### Which connector are you using (React/Angular/etc)?

React

### Bug Description

The example is forked from the reduced test case example and slimmed down. It has no interaction plug-in no dateClick actions.

Issue as reported by our accessibility scanner:
The anchor tag for the non-clickable calendar date numbers should not have an aria-label.
https://www.w3.org/TR/wai-aria-1.2/#namefromprohibited

This is an issue on: 
* Non interactive date numbers (first screenshot)
* Day of week names in the column headers (first screenshot)

Similarly on the main table with aria-labeledby="fc-dom-1" (second screenshot):
Notes from our accessibility scanner:
_Element role: generic
The div and span elements have an implicit role of generic and cannot be named unless they have a role attribute. The following roles cannot be named: caption code deletion emphasis generic insertion paragraph presentation strong subscript superscript_

Thanks!

![fullCalendar-nonClickable-removeAriaLabel](https://github.com/fullcalendar/fullcalendar/assets/12191764/7dfbb86c-6567-4081-bbe0-5f9dabf3c10d)

![fullCalendar-roleOnGeneric](https://github.com/fullcalendar/fullcalendar/assets/12191764/069c7646-0349-41d0-94aa-ac7ba3cd36fc)

**Comment by arshaw** [comment]
I've audited all instances of this problem and solved it in [v7.0.0-beta.4](https://github.com/fullcalendar/fullcalendar-workspace/releases/tag/v7.0.0-beta.4)

Updated repro:
https://codepen.io/arshaw/pen/mybXMmz?editors=0010
